### PR TITLE
Extract Percy::KeywordStruct, add rubocop config, and add documentation.

### PR DESCRIPTION
**Pull request by fotinakis**
After this PR, expect a followup that integrates this into percy-api to complete the extraction.

### CODE REVIEW

File: lib/percy/keyword_struct.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# A struct that only allows keyword arguments.
```
Comment (fotinakis) [inline lib/percy/keyword_struct.rb]: This code is extracted 1 for 1 and is not "new code" — everything else in this PR is technically "new" and should be reviewed.

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+inherit_gem:
+  percy-style:
+    - default.yml
```
Comment (djones) [inline .rubocop.yml]: So good 👍

File: lib/percy/common/version.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 module Percy
   module Common
-    VERSION = '1.1.0'
+    VERSION = '1.1.0'.freeze
```
Comment (djones) [inline lib/percy/common/version.rb]: Comment: We should consider adopting `# frozen_string_literal: true`, which would avoid the need for this call.

File: .travis.yml [hunk 4]
```diff
@@ -6,4 +6,6 @@ rvm:
   - ruby-head
 before_install:
   - gem update bundler
-script: bundle exec rspec
+script:
+  - bundle exec rubocop -D
```
Comment (timhaines) [inline .travis.yml]: yay!

**Comment by timhaines** [comment]
LGTM too.

**Comment by fotinakis** [comment]
Cool thanks for the reviews here @djones and @timhaines  👍  Let's chat about `frozen_string_literal` sometime, that sounds good to me.

**Pull request merged**